sql/sql_acl.cc

```diff
@@ -10273,7 +10273,7 @@ static int handle_grant_struct(enum enum_acl_lists struct_no, bool drop,
   /* Get the number of elements in the in-memory structure. */
   switch (struct_no) {
   case USER_ACL:
-    elements= acl_users.elements;
+    elements= int(acl_users.elements);
```

Comment:
Can you just change the `elements` variable to be size_t please. Might as well also change `idx` variable in this function as well to be `size_t`.

Reply:
The loop over the `acl_users.elements` array pushes the `idx` to `-1` and the `int elements` variable is directly manipulated as well, making the change of type of these two variables more invasive than type and casting.

Once https://github.com/MariaDB/server/pull/1930 is merged up to 10.7, then this code will be easier to change in the direction you suggest in a less invasive way.